### PR TITLE
OpenSslEngine.getSupportedCipherSuites() must return java names as well.

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

At the moment OpenSslEngine.getSupportedCipherSuites() only return the original openssl cipher names and not the java names. We need also include the java names.

Modifications:

Correctly return the java names as well.

Result:

Correct implementation of OpenSslEngine.getSupportedCipherSuites()

### CODE REVIEW

**Comment by normanmaurer** [comment]
@nmittler @Scottmitch PTAL

File: handler/src/main/java/io/netty/handler/ssl/OpenSsl.java [hunk 1]
```diff
@@ -198,6 +230,14 @@ public static Throwable unavailabilityCause() {
     }
 
     /**
+     * Returns all the available cipher suites (Java-style).
+     * Please note that the returned array may include the cipher suites that are insecure or non-functional.
+     */
+    public static Set<String> availableCipherSuitesConverted() {
```
Comment (Scottmitch) [inline handler/src/main/java/io/netty/handler/ssl/OpenSsl.java]: `availableJavaCipherSuites`?

should we consider renaming `availableCipherSuites` to `availableOpenSslCipherSuites` to future clarify (interface breaking change).

Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/ssl/OpenSsl.java]: @Scottmitch I think we could at least `@deprecated` it and add the new method like you suggested.

Comment (Scottmitch) [inline handler/src/main/java/io/netty/handler/ssl/OpenSsl.java]: sgtm

File: handler/src/main/java/io/netty/handler/ssl/OpenSsl.java [hunk 2]
```diff
@@ -104,10 +127,10 @@
                     try {
                         for (String c: SSL.getCiphers(ssl)) {
                             // Filter out bad input.
-                            if (c == null || c.length() == 0 || availableCipherSuites.contains(c)) {
+                            if (c == null || c.length() == 0 || availableOpenSslCipherSuites.contains(c)) {
```
Comment (netkins) [inline handler/src/main/java/io/netty/handler/ssl/OpenSsl.java]: MAJOR Refactor this code to not nest more than 5 if/for/while/switch/try statements. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS134)

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (7c1374dd54fda385ceb384c3d27c6a8d9bab3d10) and 4.1 (77350b7)

**Pull request closed without merging**